NPE running on Netherlands GTFS during language validation

**Summary**

When I run the current master branch as of https://github.com/MobilityData/gtfs-validator/commit/b9cd8020a7f514ce7b8ef77021fe4222091a700f on the GTFS Netherlands dataset:

http://gtfs.ovapi.nl/gtfs-nl.zip

...with:

`-i gtfs-openov-nl.zip -o output -t 8 -f nl-open`

...I get:

```
Cache for stop_times.txt departure_time: size = 77904, lookup count = 24365440, hits = 99.68%, misses = 0.32%
Cache for stop_times.txt stop_id: size = 43505, lookup count = 12182720, hits = 99.64%, misses = 0.36%
Cache for stop_times.txt stop_headsign: size = 1175, lookup count = 12182720, hits = 99.99%, misses = 0.01%
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.mobilitydata.gtfsvalidator.table.GtfsFeedLoader.lambda$loadAndValidate$3(GtfsFeedLoader.java:146)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
	at org.mobilitydata.gtfsvalidator.table.GtfsFeedLoader.loadAndValidate(GtfsFeedLoader.java:144)
	at org.mobilitydata.gtfsvalidator.cli.Main.main(Main.java:82)
Caused by: java.lang.NullPointerException
	at org.mobilitydata.gtfsvalidator.validator.MatchingFeedAndAgencyLangValidator.lambda$validate$0(MatchingFeedAndAgencyLangValidator.java:60)
Caused by: java.lang.NullPointerException

	at java.base/java.util.ArrayList.forEach(ArrayList.java:1510)
	at org.mobilitydata.gtfsvalidator.validator.MatchingFeedAndAgencyLangValidator.validate(MatchingFeedAndAgencyLangValidator.java:60)
	at org.mobilitydata.gtfsvalidator.table.GtfsFeedLoader.lambda$loadAndValidate$2(GtfsFeedLoader.java:139)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Validation took 44.547 seconds
agency.txt	29
attributions.txt	0
calendar.txt	0
calendar_dates.txt	64045
fare_attributes.txt	0
fare_rules.txt	0
feed_info.txt	1
frequencies.txt	0
levels.txt	0
pathways.txt	0
routes.txt	2561
shapes.txt	3947455
stop_times.txt	12182720
stops.txt	45087
transfers.txt	0
translations.txt	0
trips.txt	650548
```

**What I expect**

No NPE

**What I see**

NPE

**Steps to repro**

1. Run validator on Netherlands GTFS (http://gtfs.ovapi.nl/gtfs-nl.zip) with `-i gtfs-openov-nl.zip -o output -t 8 -f nl-open`

**Environment**

IntelliJ IDEA 2020.2.4 (Ultimate Edition)
Build #IU-202.8194.7, built on November 24, 2020
Runtime version: 11.0.9+11-b944.49 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1955M
Cores: 8
Non-Bundled Plugins: Lombook Plugin, idea.plugin.protoeditor, io.protostuff.protostuff-jetbrains-plugin, net.seesharpsoft.intellij.plugins.csv, org.jetbrains.kotlin

--- a/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidator.java
+++ b/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidator.java
@@ -56,7 +56,12 @@ public class MatchingFeedAndAgencyLangValidator extends FileValidator {
     Set<String> agencyLangCollection = new HashSet<>();
     agencyTable
         .getEntities()
-        .forEach(agency -> agencyLangCollection.add(agency.agencyLang().getISO3Language()));
+        .forEach(
+            agency -> {
+              if (agency.hasAgencyLang()) {
+                agencyLangCollection.add(agency.agencyLang().getISO3Language());
+              }
+            });
     if (feedInfoFeedLang.equals(Locale.forLanguageTag("mul"))) {
       // If feed_lang is mul and there isn't more than one agency_lang, that's an error
       if (agencyLangCollection.size() <= 1) {

--- a/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidatorTest.java
+++ b/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidatorTest.java
@@ -180,20 +180,21 @@ public class MatchingFeedAndAgencyLangValidatorTest {
                 "eng", new HashSet<>(Arrays.asList("ita", "fra"))));
   }
 
-    @Test
-    public void mulFeedLandAndMoreThanOneAgencyShouldNotGenerateNotice() {
-      NoticeContainer noticeContainer = new NoticeContainer();
-      MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
-      underTest.agencyTable =
-          createAgencyTable(
-              noticeContainer,
-              ImmutableList.of(
-                  createAgency(2, "agency id value", Locale.ITALIAN),
-                  createAgency(3, "other agency id value", Locale.FRANCE)));
-      underTest.feedInfoTable =
-          createFeedInfoTable(noticeContainer, ImmutableList.of(createFeedInfo(2, Locale.forLanguageTag("mul"))));
-      underTest.validate(noticeContainer);
+  @Test
+  public void mulFeedLandAndMoreThanOneAgencyShouldNotGenerateNotice() {
+    NoticeContainer noticeContainer = new NoticeContainer();
+    MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+    underTest.agencyTable =
+        createAgencyTable(
+            noticeContainer,
+            ImmutableList.of(
+                createAgency(2, "agency id value", Locale.ITALIAN),
+                createAgency(3, "other agency id value", Locale.FRANCE)));
+    underTest.feedInfoTable =
+        createFeedInfoTable(
+            noticeContainer, ImmutableList.of(createFeedInfo(2, Locale.forLanguageTag("mul"))));
+    underTest.validate(noticeContainer);
 
-      assertThat(noticeContainer.getValidationNotices()).isEmpty();
-    }
+    assertThat(noticeContainer.getValidationNotices()).isEmpty();
+  }
 }

--- a/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidatorTest.java
+++ b/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/MatchingFeedAndAgencyLangValidatorTest.java
@@ -17,20 +17,13 @@
 package org.mobilitydata.gtfsvalidator.validator;
 
 import static com.google.common.truth.Truth.assertThat;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyNoInteractions;
-import static org.mockito.Mockito.verifyNoMoreInteractions;
-import static org.mockito.Mockito.when;
 
-import java.util.ArrayList;
+import com.google.common.collect.ImmutableList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Locale;
-import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
@@ -40,32 +33,42 @@ import org.mobilitydata.gtfsvalidator.table.GtfsAgency;
 import org.mobilitydata.gtfsvalidator.table.GtfsAgencyTableContainer;
 import org.mobilitydata.gtfsvalidator.table.GtfsFeedInfo;
 import org.mobilitydata.gtfsvalidator.table.GtfsFeedInfoTableContainer;
-import org.mockito.ArgumentCaptor;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.MockitoAnnotations;
 
 @RunWith(JUnit4.class)
 public class MatchingFeedAndAgencyLangValidatorTest {
-  @Mock final GtfsFeedInfoTableContainer mockFeedInfoTable = mock(GtfsFeedInfoTableContainer.class);
-  @Mock final GtfsAgencyTableContainer mockAgencyTable = mock(GtfsAgencyTableContainer.class);
+  private static GtfsAgencyTableContainer createAgencyTable(
+      NoticeContainer noticeContainer, List<GtfsAgency> entities) {
+    return GtfsAgencyTableContainer.forEntities(entities, noticeContainer);
+  }
 
-  @InjectMocks
-  final MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+  public static GtfsAgency createAgency(long csvRowNumber, String agencyId, Locale agencyLang) {
+    return new GtfsAgency.Builder()
+        .setAgencyId(agencyId)
+        .setCsvRowNumber(csvRowNumber)
+        .setAgencyLang(agencyLang)
+        .build();
+  }
 
-  @Before
-  public void openMocks() {
-    MockitoAnnotations.openMocks(this);
+  private static GtfsFeedInfoTableContainer createFeedInfoTable(
+      NoticeContainer noticeContainer, List<GtfsFeedInfo> entities) {
+    return GtfsFeedInfoTableContainer.forEntities(entities, noticeContainer);
+  }
+
+  public static GtfsFeedInfo createFeedInfo(long csvRowNumber, Locale feedLang) {
+    return new GtfsFeedInfo.Builder()
+        .setCsvRowNumber(csvRowNumber)
+        .setFeedPublisherName("feed publisher name")
+        .setFeedPublisherUrl("www.mobilitydata.org")
+        .setFeedLang(feedLang)
+        .build();
   }
 
   @Test
   public void noFeedInfoShouldNotGenerateNotice() {
     NoticeContainer noticeContainer = new NoticeContainer();
     MatchingFeedAndAgencyLangValidator validator = new MatchingFeedAndAgencyLangValidator();
-    validator.agencyTable =
-        GtfsAgencyTableContainer.forEntities(new ArrayList<>(), noticeContainer);
-    validator.feedInfoTable =
-        GtfsFeedInfoTableContainer.forEntities(new ArrayList<>(), noticeContainer);
+    validator.agencyTable = GtfsAgencyTableContainer.forEmptyFile();
+    validator.feedInfoTable = GtfsFeedInfoTableContainer.forEmptyFile();
     validator.validate(noticeContainer);
 
     assertThat(noticeContainer.getValidationNotices()).isEmpty();
@@ -73,237 +76,124 @@ public class MatchingFeedAndAgencyLangValidatorTest {
 
   @Test
   public void mulFeedLangAndNoMoreThanOneAgencyLangShouldGenerateNotice() {
-    NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
-    when(mockFeedInfoTable.entityCount()).thenReturn(1);
+    NoticeContainer noticeContainer = new NoticeContainer();
+    MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+    underTest.agencyTable =
+        createAgencyTable(
+            noticeContainer, ImmutableList.of(createAgency(2, "agency id value", Locale.CANADA)));
+    underTest.feedInfoTable =
+        createFeedInfoTable(
+            noticeContainer, ImmutableList.of(createFeedInfo(2, Locale.forLanguageTag("mul"))));
+    underTest.validate(noticeContainer);
 
-    GtfsFeedInfo mockFeedInfo = mock(GtfsFeedInfo.class);
-    when(mockFeedInfo.feedLang()).thenReturn(Locale.forLanguageTag("mul"));
-    List<GtfsFeedInfo> feedInfoCollection = new ArrayList<>();
-    feedInfoCollection.add(mockFeedInfo);
-    when(mockFeedInfoTable.getEntities()).thenReturn(feedInfoCollection);
-
-    GtfsAgency mockAgency = mock(GtfsAgency.class);
-    when(mockAgency.agencyLang()).thenReturn(Locale.forLanguageTag("fr"));
-    List<GtfsAgency> agencyCollection = new ArrayList<>();
-    agencyCollection.add(mockAgency);
-    when(mockAgencyTable.getEntities()).thenReturn(agencyCollection);
-
-    underTest.validate(mockNoticeContainer);
-
-    final ArgumentCaptor<FeedInfoLangAndAgencyLangMismatchNotice> captor =
-        ArgumentCaptor.forClass(FeedInfoLangAndAgencyLangMismatchNotice.class);
-
-    verify(mockNoticeContainer, times(1)).addValidationNotice(captor.capture());
-
-    FeedInfoLangAndAgencyLangMismatchNotice notice = captor.getValue();
-    assertThat(notice.getCode()).matches("feed_info_lang_and_agency_lang_mismatch");
-    assertThat(notice.getContext()).containsEntry("feedInfoLang", "mul");
-    assertThat(notice.getContext())
-        .containsEntry("agencyLangCollection", new HashSet<>(Collections.singletonList("fra")));
+    assertThat(noticeContainer.getValidationNotices())
+        .containsExactly(
+            new FeedInfoLangAndAgencyLangMismatchNotice(
+                "mul", new HashSet<>(Collections.singletonList("eng"))));
   }
 
   @Test
   public void feedLangNotMulAndMoreThanOneAgencyLangShouldGenerateNotice() {
-    NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
-    when(mockFeedInfoTable.entityCount()).thenReturn(1);
+    NoticeContainer noticeContainer = new NoticeContainer();
+    MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+    underTest.agencyTable =
+        createAgencyTable(
+            noticeContainer,
+            ImmutableList.of(
+                createAgency(2, "1st agency id", Locale.CANADA),
+                createAgency(3, "2nd agency agency", Locale.FRANCE)));
+    underTest.feedInfoTable =
+        createFeedInfoTable(noticeContainer, ImmutableList.of(createFeedInfo(1, Locale.FRANCE)));
+    underTest.validate(noticeContainer);
 
-    GtfsFeedInfo mockFeedInfo = mock(GtfsFeedInfo.class);
-    when(mockFeedInfo.feedLang()).thenReturn(Locale.forLanguageTag("fr"));
-    List<GtfsFeedInfo> feedInfoCollection = new ArrayList<>();
-    feedInfoCollection.add(mockFeedInfo);
-    when(mockFeedInfoTable.getEntities()).thenReturn(feedInfoCollection);
-
-    GtfsAgency mockAgency0 = mock(GtfsAgency.class);
-    GtfsAgency mockAgency1 = mock(GtfsAgency.class);
-    when(mockAgency0.agencyLang()).thenReturn(Locale.forLanguageTag("fr"));
-    when(mockAgency1.agencyLang()).thenReturn(Locale.forLanguageTag("en"));
-    List<GtfsAgency> agencyCollection = new ArrayList<>();
-    agencyCollection.add(mockAgency0);
-    agencyCollection.add(mockAgency1);
-    when(mockAgencyTable.getEntities()).thenReturn(agencyCollection);
-
-    underTest.validate(mockNoticeContainer);
-
-    final ArgumentCaptor<FeedInfoLangAndAgencyLangMismatchNotice> captor =
-        ArgumentCaptor.forClass(FeedInfoLangAndAgencyLangMismatchNotice.class);
-
-    verify(mockNoticeContainer, times(1)).addValidationNotice(captor.capture());
-
-    FeedInfoLangAndAgencyLangMismatchNotice notice = captor.getValue();
-    assertThat(notice.getCode()).matches("feed_info_lang_and_agency_lang_mismatch");
-    assertThat(notice.getContext()).containsEntry("feedInfoLang", "fra");
-    assertThat(notice.getContext())
-        .containsEntry("agencyLangCollection", new HashSet<>(Arrays.asList("fra", "eng")));
+    assertThat(noticeContainer.getValidationNotices())
+        .containsExactly(
+            new FeedInfoLangAndAgencyLangMismatchNotice(
+                "fra", new HashSet<>(Arrays.asList("fra", "eng"))));
   }
 
   @Test
   public void feedLangNotMulAndOnlyOneMatchingAgencyLangShouldNotGenerateNotice() {
-    NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
-    when(mockFeedInfoTable.entityCount()).thenReturn(1);
+    NoticeContainer noticeContainer = new NoticeContainer();
+    MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+    underTest.agencyTable =
+        createAgencyTable(
+            noticeContainer, ImmutableList.of(createAgency(3, "2nd agency agency", Locale.FRANCE)));
+    underTest.feedInfoTable =
+        createFeedInfoTable(noticeContainer, ImmutableList.of(createFeedInfo(1, Locale.FRANCE)));
+    underTest.validate(noticeContainer);
 
-    GtfsFeedInfo mockFeedInfo = mock(GtfsFeedInfo.class);
-    when(mockFeedInfo.feedLang()).thenReturn(Locale.forLanguageTag("fr"));
-    List<GtfsFeedInfo> feedInfoCollection = new ArrayList<>();
-    feedInfoCollection.add(mockFeedInfo);
-    when(mockFeedInfoTable.getEntities()).thenReturn(feedInfoCollection);
-
-    GtfsAgency mockAgency0 = mock(GtfsAgency.class);
-    when(mockAgency0.agencyLang()).thenReturn(Locale.forLanguageTag("fr"));
-    List<GtfsAgency> agencyCollection = new ArrayList<>();
-    agencyCollection.add(mockAgency0);
-    when(mockAgencyTable.getEntities()).thenReturn(agencyCollection);
-
-    underTest.validate(mockNoticeContainer);
-
-    verifyNoInteractions(mockNoticeContainer);
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockFeedInfoTable, times(1)).entityCount();
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockFeedInfoTable, times(1)).getEntities();
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockAgencyTable, times(2)).getEntities();
-    verify(mockFeedInfo, times(1)).feedLang();
-    verify(mockAgency0, times(2)).agencyLang();
-
-    verifyNoMoreInteractions(mockFeedInfoTable, mockFeedInfo, mockAgencyTable, mockAgency0);
+    assertThat(noticeContainer.getValidationNotices()).isEmpty();
   }
 
   @Test
   public void feedLangNotMulAndOnlyOneMismatchingAgencyLangShouldGenerateNotice() {
-    NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
-    when(mockFeedInfoTable.entityCount()).thenReturn(1);
+    NoticeContainer noticeContainer = new NoticeContainer();
+    MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+    underTest.agencyTable =
+        createAgencyTable(
+            noticeContainer, ImmutableList.of(createAgency(2, "1st agency id", Locale.US)));
+    underTest.feedInfoTable =
+        createFeedInfoTable(noticeContainer, ImmutableList.of(createFeedInfo(1, Locale.FRANCE)));
+    underTest.validate(noticeContainer);
 
-    GtfsFeedInfo mockFeedInfo = mock(GtfsFeedInfo.class);
-    when(mockFeedInfo.feedLang()).thenReturn(Locale.forLanguageTag("fr"));
-    List<GtfsFeedInfo> feedInfoCollection = new ArrayList<>();
-    feedInfoCollection.add(mockFeedInfo);
-    when(mockFeedInfoTable.getEntities()).thenReturn(feedInfoCollection);
-
-    GtfsAgency mockAgency0 = mock(GtfsAgency.class);
-    when(mockAgency0.agencyLang()).thenReturn(Locale.forLanguageTag("en"));
-    List<GtfsAgency> agencyCollection = new ArrayList<>();
-    agencyCollection.add(mockAgency0);
-    when(mockAgencyTable.getEntities()).thenReturn(agencyCollection);
-
-    underTest.validate(mockNoticeContainer);
-
-    final ArgumentCaptor<FeedInfoLangAndAgencyLangMismatchNotice> captor =
-        ArgumentCaptor.forClass(FeedInfoLangAndAgencyLangMismatchNotice.class);
-
-    verify(mockNoticeContainer, times(1)).addValidationNotice(captor.capture());
-
-    FeedInfoLangAndAgencyLangMismatchNotice notice = captor.getValue();
-    assertThat(notice.getCode()).matches("feed_info_lang_and_agency_lang_mismatch");
-    assertThat(notice.getContext()).containsEntry("feedInfoLang", "fra");
-    assertThat(notice.getContext())
-        .containsEntry("agencyLangCollection", new HashSet<>(Collections.singletonList("eng")));
+    assertThat(noticeContainer.getValidationNotices())
+        .containsExactly(
+            new FeedInfoLangAndAgencyLangMismatchNotice(
+                "fra", new HashSet<>(Collections.singletonList("eng"))));
   }
 
   @Test
   public void matchingFeedInfoFeedLangShouldNotGenerateNotice() {
-    NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
-    when(mockFeedInfoTable.entityCount()).thenReturn(1);
+    NoticeContainer noticeContainer = new NoticeContainer();
+    MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+    underTest.agencyTable =
+        createAgencyTable(
+            noticeContainer,
+            ImmutableList.of(
+                createAgency(2, "agency id value", Locale.CANADA),
+                createAgency(3, "other agency id value", Locale.CANADA)));
+    underTest.feedInfoTable =
+        createFeedInfoTable(noticeContainer, ImmutableList.of(createFeedInfo(2, Locale.CANADA)));
+    underTest.validate(noticeContainer);
 
-    GtfsFeedInfo mockFeedInfo = mock(GtfsFeedInfo.class);
-    when(mockFeedInfo.feedLang()).thenReturn(Locale.forLanguageTag("fr"));
-    List<GtfsFeedInfo> feedInfoCollection = new ArrayList<>();
-    feedInfoCollection.add(mockFeedInfo);
-    when(mockFeedInfoTable.getEntities()).thenReturn(feedInfoCollection);
-
-    GtfsAgency mockAgency0 = mock(GtfsAgency.class);
-    when(mockAgency0.agencyLang()).thenReturn(Locale.forLanguageTag("fr"));
-    GtfsAgency mockAgency1 = mock(GtfsAgency.class);
-    when(mockAgency1.agencyLang()).thenReturn(Locale.forLanguageTag("fr"));
-    List<GtfsAgency> agencyCollection = new ArrayList<>();
-    agencyCollection.add(mockAgency0);
-    agencyCollection.add(mockAgency1);
-    when(mockAgencyTable.getEntities()).thenReturn(agencyCollection);
-
-    underTest.validate(mockNoticeContainer);
-
-    verifyNoInteractions(mockNoticeContainer);
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockFeedInfoTable, times(1)).entityCount();
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockFeedInfoTable, times(1)).getEntities();
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockAgencyTable, times(2)).getEntities();
-    verify(mockFeedInfo, times(1)).feedLang();
-    verify(mockAgency0, times(2)).agencyLang();
-    verify(mockAgency1, times(2)).agencyLang();
-
-    verifyNoMoreInteractions(
-        mockFeedInfoTable, mockFeedInfo, mockAgencyTable, mockAgency0, mockAgency1);
+    assertThat(noticeContainer.getValidationNotices()).isEmpty();
   }
 
   @Test
   public void feedLangNotMulAndMultipleNonMatchingAgencyLangShouldGenerateNotice() {
-    NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
-    when(mockFeedInfoTable.entityCount()).thenReturn(1);
+    NoticeContainer noticeContainer = new NoticeContainer();
+    MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+    underTest.agencyTable =
+        createAgencyTable(
+            noticeContainer,
+            ImmutableList.of(
+                createAgency(2, "agency id value", Locale.ITALIAN),
+                createAgency(3, "other agency id value", Locale.FRANCE)));
+    underTest.feedInfoTable =
+        createFeedInfoTable(noticeContainer, ImmutableList.of(createFeedInfo(2, Locale.CANADA)));
+    underTest.validate(noticeContainer);
 
-    GtfsFeedInfo mockFeedInfo = mock(GtfsFeedInfo.class);
-    when(mockFeedInfo.feedLang()).thenReturn(Locale.forLanguageTag("fr"));
-    List<GtfsFeedInfo> feedInfoCollection = new ArrayList<>();
-    feedInfoCollection.add(mockFeedInfo);
-    when(mockFeedInfoTable.getEntities()).thenReturn(feedInfoCollection);
-
-    GtfsAgency mockAgency0 = mock(GtfsAgency.class);
-    when(mockAgency0.agencyLang()).thenReturn(Locale.forLanguageTag("en"));
-    GtfsAgency mockAgency1 = mock(GtfsAgency.class);
-    when(mockAgency1.agencyLang()).thenReturn(Locale.forLanguageTag("es"));
-    List<GtfsAgency> agencyCollection = new ArrayList<>();
-    agencyCollection.add(mockAgency0);
-    agencyCollection.add(mockAgency1);
-    when(mockAgencyTable.getEntities()).thenReturn(agencyCollection);
-
-    underTest.validate(mockNoticeContainer);
-
-    final ArgumentCaptor<FeedInfoLangAndAgencyLangMismatchNotice> captor =
-        ArgumentCaptor.forClass(FeedInfoLangAndAgencyLangMismatchNotice.class);
-
-    verify(mockNoticeContainer, times(1)).addValidationNotice(captor.capture());
-
-    FeedInfoLangAndAgencyLangMismatchNotice notice = captor.getValue();
-    assertThat(notice.getCode()).matches("feed_info_lang_and_agency_lang_mismatch");
-    assertThat(notice.getContext()).containsEntry("feedInfoLang", "fra");
-    assertThat(notice.getContext())
-        .containsEntry("agencyLangCollection", new HashSet<>(Arrays.asList("spa", "eng")));
+    assertThat(noticeContainer.getValidationNotices())
+        .containsExactly(
+            new FeedInfoLangAndAgencyLangMismatchNotice(
+                "eng", new HashSet<>(Arrays.asList("ita", "fra"))));
   }
 
-  @Test
-  public void mulFeedLandAndMoreThanOneAgencyShouldNotGenerateNotice() {
-    NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
-    when(mockFeedInfoTable.entityCount()).thenReturn(1);
+    @Test
+    public void mulFeedLandAndMoreThanOneAgencyShouldNotGenerateNotice() {
+      NoticeContainer noticeContainer = new NoticeContainer();
+      MatchingFeedAndAgencyLangValidator underTest = new MatchingFeedAndAgencyLangValidator();
+      underTest.agencyTable =
+          createAgencyTable(
+              noticeContainer,
+              ImmutableList.of(
+                  createAgency(2, "agency id value", Locale.ITALIAN),
+                  createAgency(3, "other agency id value", Locale.FRANCE)));
+      underTest.feedInfoTable =
+          createFeedInfoTable(noticeContainer, ImmutableList.of(createFeedInfo(2, Locale.forLanguageTag("mul"))));
+      underTest.validate(noticeContainer);
 
-    GtfsFeedInfo mockFeedInfo = mock(GtfsFeedInfo.class);
-    when(mockFeedInfo.feedLang()).thenReturn(Locale.forLanguageTag("mul"));
-    List<GtfsFeedInfo> feedInfoCollection = new ArrayList<>();
-    feedInfoCollection.add(mockFeedInfo);
-    when(mockFeedInfoTable.getEntities()).thenReturn(feedInfoCollection);
-
-    GtfsAgency mockAgency0 = mock(GtfsAgency.class);
-    when(mockAgency0.agencyLang()).thenReturn(Locale.forLanguageTag("fr"));
-    GtfsAgency mockAgency1 = mock(GtfsAgency.class);
-    when(mockAgency1.agencyLang()).thenReturn(Locale.forLanguageTag("en"));
-    List<GtfsAgency> agencyCollection = new ArrayList<>();
-    agencyCollection.add(mockAgency0);
-    agencyCollection.add(mockAgency1);
-    when(mockAgencyTable.getEntities()).thenReturn(agencyCollection);
-
-    underTest.validate(mockNoticeContainer);
-
-    verifyNoInteractions(mockNoticeContainer);
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockFeedInfoTable, times(1)).entityCount();
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockFeedInfoTable, times(1)).getEntities();
-    //noinspection ResultOfMethodCallIgnored stubbed method
-    verify(mockAgencyTable, times(1)).getEntities();
-    verify(mockFeedInfo, times(1)).feedLang();
-    verify(mockAgency0, times(1)).agencyLang();
-    verify(mockAgency1, times(1)).agencyLang();
-
-    verifyNoMoreInteractions(
-        mockFeedInfoTable, mockFeedInfo, mockAgencyTable, mockAgency0, mockAgency1);
-  }
+      assertThat(noticeContainer.getValidationNotices()).isEmpty();
+    }
 }

